core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java

```diff
@@ -52,11 +46,16 @@ public JavaDoubleRDD union(JavaDoubleRDD... rdds) {
     if (rdds.length == 0) {
       throw new IllegalArgumentException("Union called on empty list");
     }
-    List<JavaPairRDD<K, V>> rest = new ArrayList<>(rdds.length - 1);
+    rest = populateRDDList(rdds);
+    return union(rdds[0], rest);
+  }
+  
+  public static final List<T> populateRDDList(T rdds) {
```

Comment:
This doesn't build. You're using `T` like an array type but it isn't. It has to be an array type as far as the method is concerned. `T[]` would be fine as the type.